docs/navigating-compliance-json.md

```diff
@@ -0,0 +1,87 @@
+# Navigating Compliance Using Structured Data
+
+> Warning:
+This feature is in early development and is only implemented within the API.  There are multiple caveats that need to be considered and are listed below.
+
+## Caveats
+- Must have a `Platform` created specifically for the feature.
+- Both the compliance `feature` and `rule` need to be unique for the JSON `config-type`.
+- When creating `Compliance Rules` with the config type of JSON, the `config to match` field is not used but must have bogus data in order to save to the DB.
+- Does **NOT** require Git repositories and for backup and intended.
+- Mixing/Matching CLI and JSON compliance rules is **NOT** supported.  A device should only have compliance rules from one or the other.
+
+## Quick Start Guide
+
+1. To get started with the structured data(JSON) compliance check the first step is to create a platform.
+
+![Example Platform Creation](./img/00-navigating-compliance-json.png)
+
+2. Next, the compliance feature needs to be created.
+![Example Feature Creation](./img/01-navigating-compliance-json.png)
+
+3. Link the feature that was just created to a rule definition.
+![Example Rule Creation](./img/02-navigating-compliance-json.png)
+
+> Notice: Bogus text data `none` was added to `Config to Match` which is not used in the current implementation.
+
+4. Now that the definitions are created and the rule is mapped to the platform we created the API execution can be performed.
+
+Here is an example of a `POST` to the `/plugins​/golden-config​/config-compliance​/` endpoint.
+
+```json
```

Comment:
Is this still correct with jsonfield?

Reply:
@jeffkala can you confirm?

Reply:
@itdependsnetworks trying to figure out how I get this to work.  Even with jsonfield I need to escape everything for it to work.  Maybe I need to overload the create function on the api serializer?  Trying to do some reading on what options I have.

Reply:
Still need to send it like this.
```python
data = {
  "actual": json.dumps({"ospf": ["8.8.8.8"]}),
  "intended": json.dumps({"ospf": ["8.8.8.8", "80.80.80.80"]}),
  "device": "472c3786-4499-4e34-8d8e-291c4c7f32a9",
  "rule": "475c31da-8b24-495e-aace-2ca8e57df622"
}
resp = requests.post(url, headers=h, data=json.dumps(data))
```

I believe this is correct from what I read.  @whitej6 @itdependsnetworks thoughts?

Reply:
ok, perhaps that is correct, @whitej6 do you mind confirming?

Reply:
It depends on the serializer I believe. If using jsonfield you shouldn't have to stringify the data in the payload